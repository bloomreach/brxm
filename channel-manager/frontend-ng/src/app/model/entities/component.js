/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as HstConstants from '../constants';
import { ComponentEntity } from './component-entity';

export class Component extends ComponentEntity {
  hasExperiment() {
    return !!this.getExperimentId();
  }

  getExperimentId() {
    return this._meta[HstConstants.EXPERIMENT_ID];
  }

  getExperimentStateLabel() {
    if (!this.hasExperiment()) {
      return null;
    }

    return `EXPERIMENT_LABEL_${this._meta[HstConstants.EXPERIMENT_STATE]}`;
  }

  getContainer() {
    return this.container;
  }

  setContainer(container) {
    this.container = container;
  }

  getType() {
    return 'component';
  }

  getRenderVariant() {
    return this._meta[HstConstants.RENDER_VARIANT] || HstConstants.DEFAULT_RENDER_VARIANT;
  }

  getReferenceNamespace() {
    return this._meta[HstConstants.REFERENCE_NAMESPACE];
  }
}
