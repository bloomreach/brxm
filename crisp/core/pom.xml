<?xml version="1.0" encoding="UTF-8"?>
<!--
  Copyright 2017-2018 Hippo B.V. (http://www.onehippo.com)

  Licensed under the Apache License, Version 2.0 (the  "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS"
  BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <parent>
    <groupId>org.onehippo.cms7</groupId>
    <artifactId>hippo-addon-crisp</artifactId>
    <version>14.2.2-contentfeed-SNAPSHOT</version>
  </parent>

  <name>Hippo Addon CRISP Core</name>
  <description>Hippo Addon CRISP Core</description>
  <artifactId>hippo-addon-crisp-core</artifactId>

  <dependencies>

    <dependency>
      <groupId>org.onehippo.cms7</groupId>
      <artifactId>hippo-addon-crisp-api</artifactId>
      <version>${project.version}</version>
      <scope>provided</scope>
    </dependency>

    <dependency>
      <groupId>org.onehippo.cms7</groupId>
      <artifactId>hippo-cms7-commons</artifactId>
      <scope>provided</scope>
    </dependency>

    <dependency>
      <groupId>org.onehippo.cms7</groupId>
      <artifactId>hippo-services</artifactId>
      <scope>provided</scope>
    </dependency>

    <dependency>
      <groupId>org.onehippo.cms7</groupId>
      <artifactId>hippo-repository-api</artifactId>
      <scope>provided</scope>
    </dependency>

    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-core</artifactId>
    </dependency>

    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-annotations</artifactId>
    </dependency>

    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-databind</artifactId>
    </dependency>

    <dependency>
      <groupId>org.jdom</groupId>
      <artifactId>jdom2</artifactId>
    </dependency>

    <dependency>
      <groupId>jaxen</groupId>
      <artifactId>jaxen</artifactId>
    </dependency>

    <dependency>
      <groupId>org.apache.commons</groupId>
      <artifactId>commons-lang3</artifactId>
    </dependency>

    <dependency>
      <groupId>commons-io</groupId>
      <artifactId>commons-io</artifactId>
    </dependency>

    <dependency>
      <groupId>commons-collections</groupId>
      <artifactId>commons-collections</artifactId>
    </dependency>

    <dependency>
      <groupId>org.onehippo.cms7.hst.dependencies</groupId>
      <artifactId>hst-site-dependencies</artifactId>
      <type>pom</type>
    </dependency>

    <dependency>
      <groupId>org.springframework</groupId>
      <artifactId>spring-oxm</artifactId>
    </dependency>

    <dependency>
      <groupId>org.apache.httpcomponents</groupId>
      <artifactId>httpcore</artifactId>
    </dependency>

    <dependency>
      <groupId>org.apache.httpcomponents</groupId>
      <artifactId>httpclient</artifactId>
    </dependency>

    <dependency>
      <groupId>org.freemarker</groupId>
      <artifactId>freemarker</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.security</groupId>
      <artifactId>spring-security-core</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.security</groupId>
      <artifactId>spring-security-web</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.security</groupId>
      <artifactId>spring-security-config</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.security.oauth</groupId>
      <artifactId>spring-security-oauth2</artifactId>
    </dependency>

    <!-- TEST DEPENDENCIES -->

    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <scope>test</scope>
    </dependency>

    <dependency>
      <groupId>org.easymock</groupId>
      <artifactId>easymock</artifactId>
      <scope>test</scope>
    </dependency>

    <dependency>
      <groupId>org.springframework</groupId>
      <artifactId>spring-test</artifactId>
      <scope>test</scope>
    </dependency>

  </dependencies>

</project>
