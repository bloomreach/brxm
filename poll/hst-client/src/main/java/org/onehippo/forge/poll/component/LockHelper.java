/*
 * Copyright 2009-2019 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.onehippo.forge.poll.component;

import javax.jcr.AccessDeniedException;
import javax.jcr.Node;
import javax.jcr.RepositoryException;
import javax.jcr.Workspace;
import javax.jcr.lock.LockManager;
import javax.jcr.lock.LockException;
import javax.jcr.nodetype.NodeType;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class LockHelper {
    public static final Logger logger = LoggerFactory.getLogger(LockHelper.class);

    private int timeoutMillis = 5000;
    private int sleepMillis = 250;

    public LockHelper() {
        super();
    }

    public void setTimeoutMillis(int timeoutMillis) {
        this.timeoutMillis = timeoutMillis;
    }

    public void setSleepMillis(int sleepMillis) {
        this.sleepMillis = sleepMillis;
    }


    public boolean getLock(Node node) {

        if (!makeLockable(node)) {
            return false;
        }

        boolean isLocked = false;
        long timeout = this.timeoutMillis + System.currentTimeMillis();

        while (!isLocked && (System.currentTimeMillis() < timeout)) {
            try {
                String nodePath = node.getCorrespondingNodePath(node.getSession().getWorkspace().getName());
                isLocked = doGetLock(node, nodePath);

                if (!isLocked) {
                    Thread.sleep(this.sleepMillis);
                }
            }
            catch (InterruptedException | RepositoryException e) {
                logger.error(e.getMessage());
            }
        }

        return isLocked;
    }

    public void unlock(Node node) {
        try {
            Workspace workspace = node.getSession().getWorkspace();
            LockManager lockManager = workspace.getLockManager();
            String nodePath = node.getCorrespondingNodePath(workspace.getName());

            try {
                if (node.isLocked()) {
                    lockManager.unlock(nodePath);
                }
            }
            catch (LockException le) {
                // cannot unlock node? try to undo changes and force it
                try {
                    logger.error("unable to remove the lock for node {}, undoing all changes and trying to force remove this lock", nodePath);
                    node.refresh(false);
                    lockManager.unlock(nodePath);
                }
                catch (RepositoryException e) {
                   logger.error("RepositoryException: unable to remove the lock for node " + nodePath +  ", even after undoing changes. Origination exception stacktrace:", le);
                }
            }
            catch (RepositoryException e) {
                logger.error(e.getMessage());
            }
        }
        catch (RepositoryException e) {
            logger.error(e.getMessage());
        }
    }

    private boolean doGetLock(Node node, String nodePath) {
        try {
            if (!node.isLocked()) {
                LockManager lockManager = node.getSession().getWorkspace().getLockManager();
                lockManager.lock(nodePath, true /*isDeep*/, true /*isSessionScoped*/, 5 /*timeoutHint*/, "" /*ownerInfo*/);
                return true;
            }

        }
        catch (LockException e) {
            //someone beat us to the punch, next time better luck
        }
        catch (AccessDeniedException e) {
            try {
                logger.error("Lock access denied for user {} on node {}", node.getSession().getUserID(), node.getPath());
            }
            catch (RepositoryException e1) {
                logger.error(e.getMessage());
            }
        }
        catch (Exception e) {
            try {
                logger.error("Failed setting locked lock for {}, message is {}", node.getPath(), e.getMessage());
            }
            catch (RepositoryException e1) {
                logger.error("Failed setting locked lock, message is {}",  e1.getMessage());
            }
            throw new RuntimeException(e.getMessage());
         }
        return false;
    }

    private boolean makeLockable(Node node) {
        try {
            NodeType[] nodeTypes = node.getMixinNodeTypes();
            for (NodeType nodeType : nodeTypes) {
                if ("mix:lockable".equals(nodeType.getName())) {
                    return true;
                }
            }
            if (node.canAddMixin("mix:lockable")) {
                node.addMixin("mix:lockable");
                node.getSession().save();
                node.refresh(false);
                return true;
            }
        }
        catch (RepositoryException re) {
            logger.error(re.getMessage());
        }
        return false;
    }
}
