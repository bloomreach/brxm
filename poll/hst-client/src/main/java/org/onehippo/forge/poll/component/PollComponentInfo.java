/*
 * Copyright 2012-2016 Hippo B.V. (http://www.onehippo.com)
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *  http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.onehippo.forge.poll.component;

import org.hippoecm.hst.core.parameters.JcrPath;
import org.hippoecm.hst.core.parameters.Parameter;

public interface PollComponentInfo {

    @Parameter(name = "poll-dataPath")
    String getPollDataPath();

    @Parameter(name = "poll-docsClass")
    String getPollDocsClass();

    @Parameter(name = "poll-pollCompoundName")
    String getPollCompoundName();

    @Parameter(name = "poll-docsPath", required = true)
    @JcrPath(pickerSelectableNodeTypes = "hippostd:folder")
    String getPollDocsPath();
}
