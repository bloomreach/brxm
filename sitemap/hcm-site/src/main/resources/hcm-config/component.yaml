definitions:
  config:
    /hst:hst/hst:configurations/hst:default/hst:components/forge-sitemap-feed:
      jcr:primaryType: hst:component
      hst:componentclassname: org.onehippo.forge.sitemap.components.SitemapFeed
      hst:template: forge-sitemap-feed.ftl
