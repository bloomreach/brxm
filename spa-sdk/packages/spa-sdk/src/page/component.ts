/*
 * Copyright 2019-2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { inject, injectable } from 'inversify';
import { LinkFactory } from './link-factory';
import { Link } from './link';
import { MetaCollectionFactory } from './meta-collection-factory';
import { MetaCollectionModel, MetaCollection } from './meta-collection';
import { Reference } from './reference';

export const ComponentChildrenToken = Symbol.for('ComponentChildrenToken');
export const ComponentModelToken = Symbol.for('ComponentModelToken');

/**
 * Generic component type.
 */
export const TYPE_COMPONENT = 'component';

/**
 * Container item type.
 */
export const TYPE_COMPONENT_CONTAINER_ITEM = 'container-item';

/**
 * Container type.
 */
export const TYPE_COMPONENT_CONTAINER = 'container';

export type ComponentType = typeof TYPE_COMPONENT
  | typeof TYPE_COMPONENT_CONTAINER_ITEM
  | typeof TYPE_COMPONENT_CONTAINER;

/**
 * Meta-data of a component.
 */
export interface ComponentMeta extends MetaCollectionModel {
  params?: ComponentParameters;
}

type ComponentLinks = 'self';

type ComponentModels = Record<string, any>;

type ComponentParameters = Partial<Record<string, string>>;

/**
 * Model of a component.
 */
export interface ComponentModel {
  children?: Reference[];
  id: string;
  links: Record<ComponentLinks, Link>;
  meta: ComponentMeta;
  models?: ComponentModels;
  name?: string;
  type: ComponentType;
}

/**
 * A component in the current page.
 */
export interface Component {
  /**
   * @return The component id.
   */
  getId(): string;

  /**
   * @return The component meta-data collection.
   */
  getMeta(): MetaCollection;

  /**
   * @return The map of the component models.
   */
  getModels<T extends ComponentModels>(): T;

  /**
   * @return The link to the partial component model.
   */
  getUrl(): string | undefined;

  /**
   * @return The name of the component.
   */
  getName(): string;

  /**
   * @return The parameters of the component.
   */
  getParameters(): ComponentParameters;

  /**
   * @return The direct children of the component.
   */
  getChildren(): Component[];

  /**
   * Looks up for a nested component.
   * @param componentNames A lookup path.
   */
  getComponent<U extends Component>(...componentNames: string[]): U | undefined;
  getComponent(): this;

  /**
   * Looks up for a nested component by its id.
   * @param id A component id.
   */
  getComponentById<U extends Component>(id: string): U | this | undefined;
}

@injectable()
export class ComponentImpl implements Component {
  protected meta: MetaCollection;

  constructor(
    @inject(ComponentModelToken) protected model: ComponentModel,
    @inject(ComponentChildrenToken) protected children: Component[],
    @inject(LinkFactory) private linkFactory: LinkFactory,
    @inject(MetaCollectionFactory) metaFactory: MetaCollectionFactory,
  ) {
    this.meta = metaFactory(this.model.meta);
  }

  getId() {
    return this.model.id;
  }

  getMeta() {
    return this.meta;
  }

  getModels<T extends ComponentModels>(): T;
  getModels() {
    return this.model.models || {};
  }

  getUrl() {
    return this.linkFactory.create(this.model.links.self);
  }

  getName() {
    return this.model.name || '';
  }

  getParameters() {
    return this.model.meta.params || {};
  }

  getChildren() {
    return this.children;
  }

  getComponent(): this;
  getComponent<U extends Component>(...componentNames: string[]): U | undefined;
  getComponent(...componentNames: string[]) {
    // tslint:disable-next-line:no-this-assignment
    let component: Component | undefined = this;

    while (componentNames.length && component) {
      const name = componentNames.shift()!;
      component = component.getChildren().find(component => component.getName() === name);
    }

    return component;
  }

  getComponentById<U extends Component>(id: string): U | this | undefined;
  getComponentById(id: string) {
    const queue = [this as Component];

    while (queue.length) {
      const component = queue.shift()!;

      if (component.getId() === id) {
        return component;
      }

      queue.push(...component.getChildren());
    }
  }
}

/**
 * Checks whether a value is a page component.
 * @param value The value to check.
 */
export function isComponent(value: any): value is Component {
  return value instanceof ComponentImpl;
}
