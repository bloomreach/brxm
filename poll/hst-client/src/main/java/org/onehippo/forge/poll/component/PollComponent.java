/*
 * Copyright 2009-2019 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.onehippo.forge.poll.component;

import javax.jcr.RepositoryException;
import javax.servlet.http.Cookie;

import org.hippoecm.hst.component.support.bean.BaseHstComponent;
import org.hippoecm.hst.content.annotations.Persistable;
import org.hippoecm.hst.core.component.HstComponentException;
import org.hippoecm.hst.core.component.HstRequest;
import org.hippoecm.hst.core.component.HstResponse;
import org.hippoecm.hst.core.parameters.ParametersInfo;

/**
 * Standard HST poll component that has a poll folder picker as parameter.
 */
@ParametersInfo(type = PollComponentInfo.class)
public class PollComponent extends BaseHstComponent {

    private final PollProvider pollProvider = new PollProvider();

    @Override
    public void doBeforeRender(HstRequest request, HstResponse response)
            throws HstComponentException {
        super.doBeforeRender(request, response);
        // set a cookie that is totally not important but is needed to avoid that IF a visitor does not yet have
        // any cookie, that org.onehippo.forge.poll.component.PollProvider.getPersistentValue() returns
        // org.onehippo.forge.poll.component.PollProvider.NO_COOKIE_SUPPORT
        response.addCookie(new Cookie("cookiesAllowed", "true"));
        PollComponentInfo pollComponentInfo = getComponentParametersInfo(request);
        pollProvider.doBeforeRender(request, response, pollComponentInfo);
    }

    @Persistable
    @Override
    public void doAction(HstRequest request, HstResponse response) throws HstComponentException {
        try {
            PollComponentInfo pollComponentInfo = getComponentParametersInfo(request);
            pollProvider.doAction(request, response, pollComponentInfo);
        } catch (RepositoryException e) {
            throw new HstComponentException(e);
        }
    }

}
