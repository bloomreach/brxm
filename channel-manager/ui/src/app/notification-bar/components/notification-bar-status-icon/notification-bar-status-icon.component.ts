/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Component, Input } from '@angular/core';

import { XPageStatus } from '../../../models/xpage-status.enum';

@Component({
  selector: 'em-notification-bar-status-icon',
  templateUrl: 'notification-bar-status-icon.component.html',
  styleUrls: ['notification-bar-status-icon.component.scss'],
})
export class NotificationBarStatusIconComponent {
  @Input()
  set status(value: XPageStatus) {
    this.icon = this.getIcon(value);
  }

  icon: string | undefined;

  private getIcon(status: XPageStatus): string {
    const statusIconMap = {
      [XPageStatus.Published]: 'xpage',
      [XPageStatus.Offline]: 'minus-circle-outline',
      [XPageStatus.UnpublishedChanges]: 'alert-outline',
      [XPageStatus.PublicationRequest]: 'comment-processing-outline',
      [XPageStatus.RejectedRequest]: 'comment-remove-outline',
      [XPageStatus.TakeOfflineRequest]: 'comment-processing-outline',
      [XPageStatus.ScheduledPublication]: 'calendar-clock',
      [XPageStatus.ScheduledToTakeOffline]: 'calendar-clock',
      [XPageStatus.ScheduledPublicationRequest]: 'comment-processing-outline',
      [XPageStatus.ScheduledToTakeOfflineRequest]: 'comment-processing-outline',
      [XPageStatus.ProjectInProgress]: 'minus-circle-outline',
      [XPageStatus.ProjectInReview]: 'comment-processing-outline',
      [XPageStatus.ProjectPageApproved]: 'comment-check-outline',
      [XPageStatus.ProjectPageRejected]: 'comment-remove-outline',
      [XPageStatus.ProjectRunning]: 'xpage',
      [XPageStatus.EditingSharedContainers]: 'alert-circle-outline',
      [XPageStatus.PreviousVersion]: 'alert-circle-outline',
      [XPageStatus.Locked]: 'lock-outline',
    };

    return statusIconMap[status];
  }
}
