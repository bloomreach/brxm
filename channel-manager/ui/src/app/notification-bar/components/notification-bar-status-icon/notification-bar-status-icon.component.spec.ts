/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Component, Input } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { XPageStatus } from '../../../models/xpage-status.enum';

import { NotificationBarStatusIconComponent } from './notification-bar-status-icon.component';

@Component({
  // tslint:disable-next-line:component-selector
  selector: 'mat-icon',
  template: '{{ svgIcon }}',
})
class MatIconMockComponent {
  @Input()
  svgIcon!: string;
}

describe('NotificationBarStatusIconComponent', () => {
  let component: NotificationBarStatusIconComponent;
  let fixture: ComponentFixture<NotificationBarStatusIconComponent>;

  beforeEach(() => {
    fixture = TestBed.configureTestingModule({
      declarations: [
        NotificationBarStatusIconComponent,
        MatIconMockComponent,
      ],
    }).createComponent(NotificationBarStatusIconComponent);

    component = fixture.componentInstance;
  });

  describe.each([
    ['Published', XPageStatus.Published, 'xpage'],
    ['Offline', XPageStatus.Offline, 'minus-circle-outline'],
    ['UnpublishedChanges', XPageStatus.UnpublishedChanges, 'alert-outline'],
    ['PublicationRequest', XPageStatus.PublicationRequest, 'comment-processing-outline'],
    ['RejectedRequest', XPageStatus.RejectedRequest, 'comment-remove-outline'],
    ['TakeOfflineRequest', XPageStatus.TakeOfflineRequest, 'comment-processing-outline'],
    ['ScheduledPublication', XPageStatus.ScheduledPublication, 'calendar-clock'],
    ['ScheduledToTakeOffline', XPageStatus.ScheduledToTakeOffline, 'calendar-clock'],
    ['ScheduledPublicationRequest', XPageStatus.ScheduledPublicationRequest, 'comment-processing-outline'],
    ['ScheduledToTakeOfflineRequest', XPageStatus.ScheduledToTakeOfflineRequest, 'comment-processing-outline'],
    ['ProjectInProgress', XPageStatus.ProjectInProgress, 'minus-circle-outline'],
    ['ProjectInReview', XPageStatus.ProjectInReview, 'comment-processing-outline'],
    ['ProjectPageApproved', XPageStatus.ProjectPageApproved, 'comment-check-outline'],
    ['ProjectPageRejected', XPageStatus.ProjectPageRejected, 'comment-remove-outline'],
    ['ProjectRunning', XPageStatus.ProjectRunning, 'xpage'],
    ['EditingSharedContainers', XPageStatus.EditingSharedContainers, 'alert-circle-outline'],
    ['PreviousVersion', XPageStatus.PreviousVersion, 'alert-circle-outline'],
    ['Locked', XPageStatus.Locked, 'lock-outline'],
  ])('if xpage status is %s', (statusName, statusValue, expectedIcon) => {
    beforeEach(() => {
      component.status = statusValue;

      fixture.detectChanges();
    });

    test(`should have ${expectedIcon} icon`, () => {
      expect(component.icon).toBe(expectedIcon);
    });

    test(`should render ${expectedIcon}`, () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
