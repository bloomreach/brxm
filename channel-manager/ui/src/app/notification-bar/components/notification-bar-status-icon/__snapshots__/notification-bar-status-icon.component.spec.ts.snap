// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBarStatusIconComponent if xpage status is EditingSharedContainers should render alert-circle-outline 1`] = `
<div
  id="root31"
>
  <mat-icon>
    alert-circle-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is Offline should render minus-circle-outline 1`] = `
<div
  id="root3"
>
  <mat-icon>
    minus-circle-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ProjectInProgress should render minus-circle-outline 1`] = `
<div
  id="root21"
>
  <mat-icon>
    minus-circle-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ProjectInReview should render comment-processing-outline 1`] = `
<div
  id="root23"
>
  <mat-icon>
    comment-processing-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ProjectPageApproved should render comment-check-outline 1`] = `
<div
  id="root25"
>
  <mat-icon>
    comment-check-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ProjectPageRejected should render comment-remove-outline 1`] = `
<div
  id="root27"
>
  <mat-icon>
    comment-remove-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ProjectRunning should render xpage 1`] = `
<div
  id="root29"
>
  <mat-icon>
    xpage
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is PublicationRequest should render comment-processing-outline 1`] = `
<div
  id="root7"
>
  <mat-icon>
    comment-processing-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is Published should render xpage 1`] = `
<div
  id="root1"
>
  <mat-icon>
    xpage
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is RejectedRequest should render comment-remove-outline 1`] = `
<div
  id="root9"
>
  <mat-icon>
    comment-remove-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ScheduledPublication should render calendar-clock 1`] = `
<div
  id="root13"
>
  <mat-icon>
    calendar-clock
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ScheduledPublicationRequest should render comment-processing-outline 1`] = `
<div
  id="root17"
>
  <mat-icon>
    comment-processing-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ScheduledToTakeOffline should render calendar-clock 1`] = `
<div
  id="root15"
>
  <mat-icon>
    calendar-clock
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is ScheduledToTakeOfflineRequest should render comment-processing-outline 1`] = `
<div
  id="root19"
>
  <mat-icon>
    comment-processing-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is TakeOfflineRequest should render comment-processing-outline 1`] = `
<div
  id="root11"
>
  <mat-icon>
    comment-processing-outline
  </mat-icon>
</div>
`;

exports[`NotificationBarStatusIconComponent if xpage status is UnpublishedChanges should render alert-outline 1`] = `
<div
  id="root5"
>
  <mat-icon>
    alert-outline
  </mat-icon>
</div>
`;
